programs/server/MetricsTransmitter.cpp

```diff
@@ -90,8 +88,7 @@ void MetricsTransmitter::transmit(std::vector<ProfileEvents::Count> & prev_count
         for (ProfileEvents::Event i = ProfileEvents::Event(0), end = ProfileEvents::end(); i < end; ++i)
         {
             const auto counter = ProfileEvents::global_counters[i].load(std::memory_order_relaxed);
-            const auto counter_increment = counter - prev_counters[i];
-            prev_counters[i] = counter;
+            const auto counter_increment = counter - ProfileEvents::global_counters.getAndUpdateLastValue(i, counter);
```

Comment:
Hm, so it will update last values each time we send metrics to Graphite and each time we write something to metric_log. Looks like there's a race condition, and Graphite will get previous counters written by metric_log and vice versa

Reply:
True, decided to revert changes in here and use last_values only for metric_log